cni/cni.go

```diff
@@ -249,8 +249,13 @@ func (config *NetworkConfig) GetNetworkInfo(podNamespace string) *network.Networ
 			}
 		}
 	}
+	if config.OptionalFlags.LoopbackDSR {
+		if err := hcn.DSRSupported(); err != nil {
+			return nil, err
+		}
+	}
 
-	return ninfo
```

Comment:
I am not sure what are the implications of not returning anything when the function declares that it actually returns two values. Can we follow the return pattern used in other functions e.g. getInACLRule and actually return the values from the function? Also just a suggestions, since the function errors out if the below check fails the check can be moved to the beginning of the function (i.e. no need to do anything else):

	if config.OptionalFlags.LoopbackDSR {
		if err := hcn.DSRSupported(); err != nil {
			return nil, err
		}
	}

Also please add a trace when we fail for this reason in the if block.

Reply:
A blank return is 'ok' because the method has named return parameters. Golang knows to return the values of ninfo and err.

Reply:
Thanks, I moved this to the top, added an additional trace, and am explicitly calling out the return vals (for consistency)